photon-ml/src/test/scala/com/linkedin/photon/ml/data/ScoredGameDatumTest.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data
+
+import org.testng.Assert
+import org.testng.annotations.Test
+
+import com.linkedin.photon.ml.constants.MathConst
+import com.linkedin.photon.ml.util.Utils
+
+class ScoredGameDatumTest {
+  private val epsilon = 1e-13
+
+  private val datumZeroScoreOneResponse = ScoredGameDatum(1.0, 0.0, 1.0, 0.0, Map())
+  private val customDatum = ScoredGameDatum(2.3, 0.5, 1.2, 11.0, Map("src" -> "1234", "dst" -> "5678"))
+  private val similarCustomDatum = ScoredGameDatum(
+    2.3 + epsilon,
+    0.5 - epsilon,
+    1.2 + epsilon,
+    -2.0,
+    Map("src" -> "1234", "dst" -> "5678"))
+  private val customDatumWithNanResponse = ScoredGameDatum(
+    Double.NaN,
+    0.5,
+    1.2,
+    11.0,
+    Map("src" -> "1234", "dst" -> "5678"))
+  private val similarCustomDatumWithNanResponse = ScoredGameDatum(
+    Double.NaN,
+    0.5 + epsilon,
+    1.2 - epsilon,
+    -2.0,
+    Map("src" -> "1234", "dst" -> "5678"))
+
+  @Test
+  def zeroScoreDatumTest(): Unit = {
```

Comment:
Open up any other unit/integration test class and every `@Test` annotated method begins with "test".